c/tskit/trees.h

```diff
@@ -362,42 +363,42 @@ int tsk_treeseq_f4(tsk_treeseq_t *self, tsk_size_t num_sample_sets,
 int tsk_tree_init(tsk_tree_t *self, tsk_treeseq_t *tree_sequence, tsk_flags_t options);
 int tsk_tree_free(tsk_tree_t *self);
 
-tsk_id_t tsk_tree_get_index(tsk_tree_t *self);
-tsk_size_t tsk_tree_get_num_roots(tsk_tree_t *self);
+tsk_id_t tsk_tree_get_index(const tsk_tree_t *self);
+tsk_size_t tsk_tree_get_num_roots(const tsk_tree_t *self);
 
 int tsk_tree_first(tsk_tree_t *self);
 int tsk_tree_last(tsk_tree_t *self);
 int tsk_tree_next(tsk_tree_t *self);
 int tsk_tree_prev(tsk_tree_t *self);
 int tsk_tree_clear(tsk_tree_t *self);
 
-void tsk_tree_print_state(tsk_tree_t *self, FILE *out);
+void tsk_tree_print_state(const tsk_tree_t *self, FILE *out);
 /** @} */
 
 int tsk_tree_set_root_threshold(tsk_tree_t *self, tsk_size_t root_threshold);
-tsk_size_t tsk_tree_get_root_threshold(tsk_tree_t *self);
+tsk_size_t tsk_tree_get_root_threshold(const tsk_tree_t *self);
 
-bool tsk_tree_has_sample_lists(tsk_tree_t *self);
-bool tsk_tree_has_sample_counts(tsk_tree_t *self);
-bool tsk_tree_equals(tsk_tree_t *self, tsk_tree_t *other);
+bool tsk_tree_has_sample_lists(const tsk_tree_t *self);
+bool tsk_tree_has_sample_counts(const tsk_tree_t *self);
+bool tsk_tree_equals(const tsk_tree_t *self, const tsk_tree_t *other);
 /* Returns true if u is a descendant of v; false otherwise */
-bool tsk_tree_is_descendant(tsk_tree_t *self, tsk_id_t u, tsk_id_t v);
-bool tsk_tree_is_sample(tsk_tree_t *self, tsk_id_t u);
+bool tsk_tree_is_descendant(const tsk_tree_t *self, tsk_id_t u, tsk_id_t v);
+bool tsk_tree_is_sample(const tsk_tree_t *self, tsk_id_t u);
 
 int tsk_tree_copy(tsk_tree_t *self, tsk_tree_t *dest, tsk_flags_t options);
```

Comment:
Copy can have ``const self`` I think?

Reply:
Ooo, yes, this one can get const early in this PR. I'll tweak and merge main and push an update.